d2common/d2math/vector2.go

```diff
@@ -0,0 +1,367 @@
+	return v.x.Cmp(src.X()) == 0 && v.y.Cmp(src.Y()) == 0
+}
+
+// FuzzyEquals checks if the Vector2 is approximately equal
+// to the given Vector2. epsilon is what we consider `smol enough`
+func (v *Vector2) FuzzyEquals(src d2interface.Vector2) bool {
+	smol := big.NewFloat(epsilon)
+	d := v.Distance(src)
+	d.Abs(d)
+
+	return d.Cmp(smol) < 1 || d.Cmp(smol) < 1
+}
+
+// Abs returns a clone that is positive
+func (v *Vector2) Abs() d2interface.Vector2 {
+	clone := v.Clone()
+	neg1 := big.NewFloat(-1.0)
+
+	if clone.X().Sign() == -1 { // is negative1
+		clone.X().Mul(clone.X(), neg1)
+	}
+
+	if v.Y().Sign() == -1 { // is negative1
+		clone.Y().Mul(clone.Y(), neg1)
+	}
+
+	return clone
+}
+
+// Angle computes the angle in radians with respect
+// to the positive x-axis
+func (v *Vector2) Angle() *big.Float {
+	// HACK we should find a way to do this purely
+	// with big.Float
+	floatX, _ := v.X().Float64()
+	floatY, _ := v.Y().Float64()
+	floatAngle := math.Atan2(floatX, floatY)
+
+	if floatAngle < 0 {
+		floatAngle += 2.0 * math.Pi
+	}
+
+	return big.NewFloat(floatAngle)
+}
+
+// SetAngle sets the angle of this Vector2
+func (v *Vector2) SetAngle(angle *big.Float) d2interface.Vector2 {
+	return v.SetToPolar(angle, v.Length())
+}
+
+// Add to this Vector2 the components of the given Vector2
+func (v *Vector2) Add(src d2interface.Vector2) d2interface.Vector2 {
+	v.x.Add(v.x, src.X())
+	v.y.Add(v.y, src.Y())
+
+	return v
+}
+
+// Subtract from this Vector2 the components of the given Vector2
+func (v *Vector2) Subtract(src d2interface.Vector2) d2interface.Vector2 {
+	v.x.Sub(v.x, src.X())
+	v.y.Sub(v.y, src.Y())
+
+	return v
+}
+
+// Multiply this Vector2 with the components of the given Vector2
+func (v *Vector2) Multiply(src d2interface.Vector2) d2interface.Vector2 {
+	v.x.Mul(v.x, src.X())
+	v.y.Mul(v.y, src.Y())
+
+	return v
+}
+
+// Scale this Vector2 by the given value
+func (v *Vector2) Scale(s *big.Float) d2interface.Vector2 {
+	v.x.Sub(v.x, s)
+	v.y.Sub(v.y, s)
+
+	return v
+}
+
+// Divide this Vector2 by the given Vector2
+func (v *Vector2) Divide(src d2interface.Vector2) d2interface.Vector2 {
+	v.x.Quo(v.x, src.X())
+	v.y.Sub(v.y, src.Y())
+
+	return v
+}
+
+// Negate thex and y components of this Vector2
+func (v *Vector2) Negate() d2interface.Vector2 {
+	return v.Scale(big.NewFloat(negative1))
+}
+
+// Distance calculate the distance between this Vector2 and the given Vector2
+func (v *Vector2) Distance(src d2interface.Vector2) *big.Float {
+	dist := v.DistanceSq(src)
+
+	return dist.Sqrt(dist)
+}
+
+// DistanceSq calculate the distance suared between this Vector2 and the given
+// Vector2
+func (v *Vector2) DistanceSq(src d2interface.Vector2) *big.Float {
+	delta := src.Clone().Subtract(v)
+	deltaSq := delta.Multiply(delta)
+
+	return big.NewFloat(zero).Add(deltaSq.X(), deltaSq.Y())
+}
+
+// Length returns the length of this Vector2
+func (v *Vector2) Length() *big.Float {
+	xsq, ysq := v.LengthSq()
```

Comment:
(Maybe I don't get what this function does.) It probably doesn't matter, but I think this is called a `norm` or Euclidian distance, and not a length. On the other hand, people will 100% understand the term length.
If this function is supposed to return the norm of a vector, it's missing a square root.